samples/joke/src/main/java/io/quarkiverse/operatorsdk/samples/joke/JokeReconciler.java

```diff
@@ -16,76 +16,16 @@
 
 import static io.javaoperatorsdk.operator.api.reconciler.Constants.WATCH_CURRENT_NAMESPACE;
 
-import java.util.Arrays;
-import java.util.stream.Collectors;
-
-import javax.inject.Inject;
-
-import org.eclipse.microprofile.rest.client.inject.RestClient;
-
-import io.fabric8.kubernetes.client.KubernetesClient;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
-import io.quarkiverse.operatorsdk.samples.joke.JokeRequestSpec.ExcludedTopic;
-import io.quarkiverse.operatorsdk.samples.joke.JokeRequestStatus.State;
 
 @ControllerConfiguration(namespaces = WATCH_CURRENT_NAMESPACE)
-public class JokeReconciler implements Reconciler<JokeRequest> {
-    @Inject
-    @RestClient
-    JokeService jokes;
-
-    @Inject
-    KubernetesClient client;
+public class JokeReconciler implements Reconciler<Joke> {
```

Comment:
I had to add a dummy controller to workaround this issue: https://github.com/quarkiverse/quarkus-operator-sdk/issues/90

Reply:
I don't think that this should be done, at least not for the purpose of putting the operator into prod: in this somewhat contrived scenario, the `Joke` CRD is assumed to be installed on the cluster. It is not technically a CR type that this operator owns so it shouldn't be part of its installation process. It should, however, be part of its requirements for installation, which, as far as I'm aware, OLM supports.

Reply:
I tried the fabric8 `apt` dependency and didn't work. 
We might argue that Joke CRD might own to the operator or not. For me, it does and we should package it.
But the other problem is that the service account that the operator is generating, it's only seeing joke requests, not jokes. So, when creating a new Joke, it fails.

Therefore, either (1) we update this example to get rid of this controversial Joke CRD and populate the joke in the JokeRequest status, (2) or we find a way that users can register and package CRDs that are not directly used by their operator.

Wdyt?

Reply:
This sample simulates a real-world scenario (albeit somewhat awkwardly) where your operator needs to integrate with a 3rd party operator (in this case, one that would handle the `Joke` CR). Granted, it does a poor job at it but it has at least the benefit of being not too complex to understand. 
In this scenario, our operator doesn't own the `Joke` resources and, in a real world scenario, that CRD would be provided for us to use (or retrieve from the cluster). Ideally, the `Joke` class would actually be generated from the CRD (which is now feasible using the fabric8 crd-to-java generator). This could actually be a future evolution of this sample. For the time being, though, the generated RBACs need to allow permissions on the `Joke` resources somehow and the `Joke` CRD should not be part of the OLM manifest for this operator but rather defined as a requirement (I know it's possible to do, I haven't looked at how OLM implements this feature, though).

Reply:
I think this comment has been already addressed. Marked as resolved.